<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

## Excercise 1

In [1]:
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

## Excercise 2

In [2]:
print = 'PRINT'

In [3]:
print('Hello World')

TypeError: 'str' object is not callable

In [1]:
print('Hello Wrold')

Hello Wrold


## Excercise 3

In [24]:
%%timeit
squares = [i**2 for i in range(10)]

2.23 μs ± 98.1 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)


In [27]:
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [26]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

2.91 μs ± 140 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)


In [23]:
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


Discussion:
The first code took 2.23 μs to execute, while the second code took longer, 2.91 μs, to execute, even though both codes give the same result. This may be because the second code has more instructions, thus it would take longer for the command to be executed. Another reason may be that the first code is more efficient than the second to be executed.